### PR TITLE
Add project issues API.

### PR DESCRIPTION
**Pull request by tmcgilchrist**

### CODE REVIEW

File: lib/gitlab.atd [hunk 1]
```diff
@@ -483,14 +492,81 @@ type author = {
...
+  title: string;
+  description: string;
+  state: state;
+  created_at: date_time;
+  updated_at: date_time;
+  closed_at: string nullable;
+  closed_by: string nullable;
+  labels: string list;
+  milestone: string nullable;
+  assignees: int list;
```
Comment (maiste) [inline lib/gitlab.atd:536]: Isn't assignees a `user_short` list?

File: lib/gitlab_core.ml [hunk 2]
```diff
@@ -1478,6 +1486,21 @@ struct
         API.post ~token ~uri ~body ~expected_code:`Created (fun s ->
             Lwt.return (Gitlab_j.project_access_token_of_string s))
     end
+
+    module Issue = struct
+      let list ?token ~project_id () =
+        let uri = URI.project_issues project_id in
+        API.get_stream ?token ~uri (fun body -> return (Gitlab_j.issues_of_string body))
```
Comment (maiste) [inline lib/gitlab_core.ml:1493]: Don't you want to force the user to give the `token` here?

File: lib/gitlab_core.ml [hunk 3]
```diff
@@ -1478,6 +1486,21 @@ struct
         API.post ~token ~uri ~body ~expected_code:`Created (fun s ->
             Lwt.return (Gitlab_j.project_access_token_of_string s))
     end
+
+    module Issue = struct
+      let list ?token ~project_id () =
```
Comment (maiste) [inline lib/gitlab_core.ml:1491]: What do you think about adding filters here as in the `merge_requests` function?

**Review by maiste — COMMENTED** [review]
The tests are working. Just a few questions but otherwise LGTM :+1:

File: lib/gitlab_core.ml (hunk 2 above)
Comment (tmcgilchrist) [reply lib/gitlab_core.ml:1493]: In this case no, you can still list the issues on a project without a token, you'll get the publicly available issues and a worse rate limit on HTTP requests.

File: lib/gitlab_core.ml (hunk 3 above)
Comment (tmcgilchrist) [reply lib/gitlab_core.ml:1491]: Great idea, there's already some examples of how to do that.

**Comment by tmcgilchrist** [comment]
Superseded by https://github.com/tmcgilchrist/ocaml-gitlab/pull/52

**Pull request closed without merging**